org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/tests/test.js

```diff
@@ -0,0 +1,65 @@
+/*jshint esversion: 6 */
+var opent2t = require('opent2t');
+var ava = require('ava');
+var config = require('./testconfig');
+var delay = require('delay')
+
+// This boilerplate code is copied from transpiled typescript AVA test code, which enables usage of 'yield' keyword. 
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {

```

Comment:
This is a great discussion, but I'm hoping we don't need to block this PR for this. @sanjaiganesh I added issue #55 to track this... perhaps you could take a stab at it? For this PR I think this is #WontFix and we can clean it up later (very soon hopefully, perhaps tomorrow/friday).